ui/components/ShipBuildingQueue.tsx

```diff
@@ -10,8 +10,20 @@ import { useGameDialog, useSim } from "@ui/atoms";
 import { relationThresholds } from "@core/components/relations";
 import redoIcon from "@assets/ui/redo.svg";
 import SVG from "react-inlinesvg";
+import type { Faction } from "@core/archetypes/faction";
+import type { ShipyardQueueItem } from "@core/components/shipyard";
 import styles from "./ShipBuildingQueue.scss";
 
+const getQueueItemOwner = (
+  entity: RequireComponent<"shipyard">,
+  queueItem: ShipyardQueueItem
+): string => {
+  const owner = queueItem.owner
+    ? entity.sim.getOrThrow<Faction>(queueItem.owner)
+    : false;
+  return owner && owner.cp.name.slug ? owner.cp.name.slug : "???";
```

Comment:
Owner is always there so no need for checking for it. And if it isn't, it's probably good to crash and know there is a bug (because we don't have any bug tracker yet ;) )

Reply:
updated